packages/react/src/components/Tabs/next/Tabs.js

```diff
@@ -194,26 +314,124 @@ function TabList({
     }
   });
 
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setIsScrollable(ref.current.scrollWidth > ref.current.clientWidth);
+    }
+
+    function handler() {
+      if (ref.current) {
+        setIsScrollable(ref.current.scrollWidth > ref.current.clientWidth);
+      }
+    }
+
+    const debouncedHandler = debounce(handler, 200);
+    window.addEventListener('resize', debouncedHandler);
+    return () => {
+      debouncedHandler.cancel();
+      window.removeEventListener('resize', debouncedHandler);
+    };
+  }, []);
+
+  // updates scroll location for all scroll behavior.
+  useLayoutEffect(() => {
+    ref.current.scrollLeft = scrollLeft;
+  }, [scrollLeft]);
+
+  // Hook to control onClick, onMouseDown, and onMouse up scroll behavior for scrolling to the right.
+  usePressable(nextButton, {
+    onPress() {
+      setNextPressed(true);
+    },
+    onPressed() {
+      const overflowInterval = setInterval(() => {
+        setScrollLeft((scrollLeft) => scrollLeft + 10);
+      });
+      return () => {
+        clearInterval(overflowInterval);
+      };
+    },
+    onRelease() {
+      if (nextPressed) {
+        setNextPressed(false);
+        return;
+      }
+
+      setScrollLeft(
+        Math.min(
+          scrollLeft + (ref.current.scrollWidth / tabs.length) * 1.5,
+          ref.current.scrollWidth - ref.current.clientWidth
+        )
+      );
+    },
+  });
+
+  // Hook to control onClick, onMouseDown, and onMouse up scroll behavior for scrolling to the left.
+  usePressable(previousButton, {
+    onPress() {
+      setPreviousPressed(true);
+    },
+    onPressed() {
+      const overflowInterval = setInterval(() => {
+        setScrollLeft((scrollLeft) => scrollLeft - 10);
+      });
+      return () => {
+        clearInterval(overflowInterval);
+      };
+    },
+    onRelease() {
+      if (previousPressed) {
+        setPreviousPressed(false);
+        return;
+      }
+
+      setScrollLeft(
+        Math.max(scrollLeft - (ref.current.scrollWidth / tabs.length) * 1.5, 0)
+      );
+    },
+  });
+
   return (
-    // eslint-disable-next-line jsx-a11y/interactive-supports-focus
-    <div
-      {...rest}
-      aria-label={label}
-      ref={ref}
-      role="tablist"
-      className={className}
-      onKeyDown={onKeyDown}>
-      {React.Children.map(children, (child, index) => {
-        const ref = React.createRef();
-        tabs.push(ref);
-        return (
-          <TabContext.Provider value={index}>
-            {React.cloneElement(child, {
-              ref,
-            })}
-          </TabContext.Provider>
-        );
-      })}
+    <div className={className}>
+      <button
+        aria-hidden="true"
+        aria-label="Scroll left"
```

Comment:
Is the idea that `aria-label` would be configurable through `leftOverflowButtonProps` or is there another way we want people to customize this label?